hi-ml-histopathology/src/histopathology/datamodules/base_module.py

```diff
@@ -150,40 +201,139 @@ def _load_dataset(self, tiles_dataset: TilesDataset, stage: str, shuffle: bool)
         # Dataset is saved if cache_dir is True, regardless of CacheMode
         if dataset_pickle_path:
             dataset_pickle_path.parent.mkdir(parents=True, exist_ok=True)
-            with dataset_pickle_path.open('wb') as f:
+            with dataset_pickle_path.open("wb") as f:
                 torch.save(transformed_bag_dataset, f)
 
         return transformed_bag_dataset
 
-    def _get_transformed_dataset(self, base_dataset: BagDataset,
-                                 transform: Union[Sequence[Callable], Callable]) -> Dataset:
-        if self.cache_mode is CacheMode.MEMORY:
-            dataset = CacheDataset(base_dataset, transform, num_workers=1)  # type: ignore
-        elif self.cache_mode is CacheMode.DISK:
-            dataset = PersistentDataset(base_dataset, transform, cache_dir=self.cache_dir)  # type: ignore
-            if self.precache_location != CacheLocation.NONE:
-                import tqdm  # TODO: Make optional
-
-                for i in tqdm.trange(len(dataset), desc="Loading dataset"):
-                    dataset[i]  # empty loop to pre-compute all transformed samples
-        else:
-            dataset = Dataset(base_dataset, transform)  # type: ignore
-        return dataset
-
-    def _get_dataloader(self, tiles_dataset: TilesDataset, stage: str, shuffle: bool,
-                        **dataloader_kwargs: Any) -> DataLoader:
-        transformed_bag_dataset = self._load_dataset(tiles_dataset, stage=stage, shuffle=shuffle)
+    def _get_dataloader(
+        self, dataset: TilesDataset, stage: str, shuffle: bool, **dataloader_kwargs: Any
+    ) -> DataLoader:
+        transformed_bag_dataset = self._load_dataset(dataset, stage=stage, shuffle=shuffle)
         bag_dataset: BagDataset = transformed_bag_dataset.data  # type: ignore
         generator = bag_dataset.bag_sampler.generator
-        return DataLoader(transformed_bag_dataset, batch_size=self.batch_size,
-                          collate_fn=multibag_collate, shuffle=shuffle, generator=generator,
-                          **dataloader_kwargs)
+        return DataLoader(
+            transformed_bag_dataset,
+            batch_size=self.batch_size,
+            collate_fn=multibag_collate,
+            shuffle=shuffle,
+            generator=generator,
+            pin_memory=False,  # disable pinning as loaded data may already be on GPU
```

Comment:
Just read through this thread here https://stackoverflow.com/questions/55563376/pytorch-how-does-pin-memory-work-in-dataloader, after which I'd say to always pin memory, and the comment is also not needed